app/app.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from flask import Flask, jsonify, Response
+import redis
+
+
+app = Flask(__name__)
+slave_replicas = int(os.getenv("REDIS_SLAVE_REPLICAS", 1))
+write_timeout = int(os.getenv("REDIS_WRITE_TIMEOUT", 10)) * 1000
+
+
+@app.route("/api/data/<key>")
+def key(key):
+    r = redis.StrictRedis(host="redis-slave", port=6379, db=0)
+
+    value = r.get(key)
+    return jsonify({"data": value.decode("ascii") if value else ""})
+
+
+@app.route("/api/data/<key>/<value>", methods=["POST"])
+def set_key(key, value):
+    r = redis.StrictRedis(host="redis-master", port=6379, db=0)
+
+    ret = r.set(key, value)
+    if not ret:
+        raise Exception("Value not setted")
+    ret = r.wait(slave_replicas, write_timeout)
```

Comment:
FYI this line makes the writes fully synchronous (https://redis.io/commands/wait)